### PR TITLE
remove `initialFocus` from `Popover`

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This removes the explicit [`initialFocus`](https://floating-ui.com/docs/FloatingFocusManager#initialfocus) prop which was set to the popover itself, because it sometimes conflicts with focus management inside the popover content.

### After-PR TODOs

- [ ] Add a `Popover.Heading` component that gets automatically focused. See [discussion](https://github.com/iTwin/iTwinUI/pull/2168#discussion_r1697270691).

## Testing

Tested that a `Popover` without any interactive elements continues to get focused when opened using keyboard, and that a `Popover` containing interactive elements gives priority to the elements inside it. Added e2e tests to confirm this.

Updated any failing test images.

## Docs

Added changeset.

The focus behavior of `Popover` is [already documented](https://itwinui.bentley.com/docs/popover#accessibility).

### CODE REVIEW

**Review by Ben-Pusey-Bentley — APPROVED** [review]

File: testing/e2e/app/routes/Popover/route.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
```
[the remark this answers is not in the document]
Comment (mayank99) [reply testing/e2e/app/routes/Popover/route.tsx]: Removed ([`f1ecb1b`](https://github.com/iTwin/iTwinUI/pull/2168/commits/f1ecb1b218a2f6f51286d926173ec8adca398a96)). It was leftover from a previous version where I was using some hooks

File: testing/e2e/app/routes/Popover/spec.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('Popover (focus)', () => {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply testing/e2e/app/routes/Popover/spec.ts:3]: It should prioritize imperative focus over everything else. If they want to focus the dialog, they can do so imperatively.

I added a combined e2e test ([`e88ae10`](https://github.com/iTwin/iTwinUI/pull/2168/commits/e88ae103423be1829d9f58ef147559bc177c5954)) but it's currently failing, even though the behavior is correct when I test it manually. Looking into this.

[the remark this answers is not in the document]
Comment (mayank99) [reply testing/e2e/app/routes/Popover/spec.ts:3]: I answered this above: "If they want to focus the dialog, they can do so imperatively."

It is currently the application developer's responsibility to move focus to the right place. Neither the popover nor the combobox are great initial focus points; ideally the focus should move to the first heading at the top of the `Popover`.

Maybe in the future we can provide a `Popover.Heading` component that adds the recommended markup/visuals and gets automatically focused.

[the remark this answers is not in the document]
Comment (mayank99) [reply testing/e2e/app/routes/Popover/spec.ts:3]: Added after-PR TODO.

One more thing I wanted to mention: the current behavior matches how the native `<dialog>` element works: it looks for the first focusable element.




<details><summary>Code</summary>

```html
<button onclick="dialog.showModal()">Open dialog</button>
<dialog id="dialog">
  <h2>This is the dialog</h2>
  <input placeholder="Pretend this is a combobox" />
</dialog>
```

</details>

**Pull request merged**